### DISCUSSION TITLE
Using local JSON file as a content source

### DISCUSSION THREAD

**Discussion by spryzone** [vikejs/vike#932] [Discussion] [answered]
Using Vite/React/vite-plugin-ssr, how would I prerender pages for SSG using a local static JSON file as the source, e.g.:

```
{
  "posts": [
    {
      "id": "post-1",
      "author": "John",
      "title": "Post 1",
      "content": "<p>Some content</p>"
    },
    {
      "id": "post-2",
      "author": "John",
      "title": "Post 2",
      "content": "<p>Some content...</p>"
    }
  ],
  "authors": [
    {
      "id": "john",
      "name": "John"
    }
  ]
}
```

I have looked through the docs, but I can't see an example using a local JSON file. I don't need to fetch any remote content, e.g. https://github.com/brillout/vite-plugin-ssr/blob/main/examples/react-full/pages/star-wars/index/index.page.server.ts and don't use MDX, only JSON

**Comment by brillout** [comment, marked as answer]
Why not `onBeforeRender()` as usual?

**Reply by spryzone** [reply]
Thanks, that did the trick. I was over thinking it!
